shell/packages/sandstorm-permissions/permissions.js

```diff
@@ -399,136 +433,447 @@ class Context {
     }
   }
 
-  addGrain(db, grainId) {
-    check(db, SandstormDb);
-    check(grainId, String);
-    const grain = db.getGrain(grainId);
-    this.grains[grainId] = grain;
+  addGrains(db, grainIds) {
```

Comment:
It looks like various callers of this method could conceivably pass in an empty grainIds, in which case the database query is a waste.